### ISSUE TITLE
DetailsList - in SelectionMode.none - able to select row using mouse, unable to using keyboard

### ISSUE THREAD

**Issue by eveleens** [OfficeDev/office-ui-fabric-react#8543]
## Codepen demonstrating the issue
Turn off modal selection in: https://developer.microsoft.com/en-us/fabric#/components/detailslist

## Actual behavior:
You can use your mouse to make a row selection. You can NOT push space or enter to make a row selection.

## Expected behavior:
Unclear, but it should be consistent. 

I have a case where I'd like to be able to select without showing the checkmarks, so what currently exists works for me for mouse users. However, it doesn't work for keyboard users. 

Though, I presume the fact that mouse users can make a selection is a bug when we explicitly set the selectionMode to none.

**Comment by JoshuaLamusga** [comment]
I can confirm this behavior is occurring.

**Comment by KevinTCoughlin** [comment]
@betrue-final-final, is it a bug that `DetailsList` allows for item selection via "Click" while modal selection is set to "none"? The behavior is exhibited at https://developer.microsoft.com/en-us/fabric#/components/detailslist.

**Comment by betrue-final-final** [comment]
A keyboard user MUST be able to use space to select an item. This is definitely a bug.

**Comment by KevinTCoughlin** [comment]
After chatting with @betrue-final-final, it seems like we at least want parity with keyboard & mouse in the `SelectionMode === None` case.

Currently in that case, users can only "click" and select _a single_ item. Keyboard users cannot. We'll need to chat with @ThomasMichon to see do we either:

1) Disable single selection entirely in this mode

OR 

2) Keep "click" behavior and allow for `SPACE` to perform the same interaction.

The naming makes me think we want to remove the behavior, but we'd need to see what the OneDrive / SharePoint implications are.

**Comment by dzearing** [comment]
When `SelectonMode === None`, selection should not work, period.

**Comment by KevinTCoughlin** [comment]
> When `SelectonMode === None`, selection should not work, period.

I have a fix for this, just need to add tests and communicate the change.

**Comment by KevinTCoughlin** [comment]
@eveleens, I've provided a potential solution and alternative in https://github.com/OfficeDev/office-ui-fabric-react/pull/8605. Thanks for raising awareness to this issue.

The scope of the proposed fix impacts `SelectionZone` across OUFR cc: @betrue-final-final @dzearing
